src/jit/lsra.cpp

```diff
@@ -7952,6 +7952,17 @@ void LinearScan::resolveLocalRef(BasicBlock* block, GenTreePtr treeNode, RefPosi
         if (interval->assignedReg != nullptr && interval->assignedReg->assignedInterval == interval)
         {
             interval->assignedReg->assignedInterval = nullptr;
+#ifdef _TARGET_ARM_
+            // Update overlapping floating point register for TYP_DOUBLE
+            if (interval->registerType == TYP_DOUBLE)
+            {
+                assert(genIsValidDoubleReg(interval->assignedReg->regNum));
+
+                RegRecord* nextPhysRegRecord = findAnotherHalfRegRec(interval->assignedReg);
+
+                nextPhysRegRecord->assignedInterval = nullptr;
+            }
+#endif
```

Comment:
@CarolEidt You are right and I'm planning to do some refactoring for `lsra.cpp`.

For example, we can reuse part of `checkAndClearInterval` and `checkAndAssignInterval` when updating `assignedInterva`l in `RegRecord`. If you don't mind I will make another PR for refactoring after many bugs in register allocation for ryujit/arm32 are fixed.